TDDWorkshop/Classes/PhotoStream/PhotoStreamViewController.swift

```diff
@@ -92,26 +96,26 @@ class PhotoStreamViewController: UICollectionViewController, ItemCreatingDelegat
         }
     }
 
-    func creator(creator: ItemCreating, failedWithError: ErrorType) {
+    func creator(_ creator: ItemCreating, failedWithError: Error) {
         presentErrorAlertWithMessage("Failed to create stream item!")
     }
 
     //MARK: Private methods
 
-    private func presentErrorAlertWithMessage(message: String) {
-        let errorAlert = UIAlertController(title: "Error", message: message, preferredStyle: .Alert)
-        errorAlert.addAction(alertActionFactory.createActionWithTitle("Cancel", style: .Cancel) {
+    fileprivate func presentErrorAlertWithMessage(_ message: String) {
+        let errorAlert = UIAlertController(title: "Error", message: message, preferredStyle: .alert)
+        errorAlert.addAction(alertActionFactory.createActionWithTitle("Cancel", style: .cancel) {
             action in })
         presenter.presentViewController(errorAlert)
     }
 
-    private func setupCollectionView() {
-        refreshControl.addTarget(self, action: #selector(didPullToRefresh), forControlEvents: UIControlEvents.ValueChanged)
+    fileprivate func setupCollectionView() {
+        refreshControl.addTarget(self, action: #selector(didPullToRefresh), for: UIControlEvents.valueChanged)
         collectionView?.addSubview(refreshControl)
         collectionView?.alwaysBounceVertical = true
     }
 
-    private func downloadStreamItems() {
+    fileprivate func downloadStreamItems() {
```

Comment:
This is automatic Swift 3 migration :) I would take a closer look when updating PhotoStream part.